SygicUIKit/Classes/ColorSchemes/SYUIColorSchemeManager.swift

```diff
@@ -0,0 +1,71 @@
+import Foundation
+
+@objc public protocol SYUIColorUpdate {
+    @objc func setupColors()
+}
+
+public enum ColorScheme {
+    case day
+    case night
+}
+
+public let ColorPaletteChangedNotification: String = "ColorPaletteChangedNotification"
+
+public class SYUIColorSchemeManager: NSObject {
+
+    public static let shared = SYUIColorSchemeManager()
+    
+    public var currentColorPalette: SYUIColorPalette = SYUIDefaultColorPalette() {
+        didSet {
+            NotificationCenter.default.post(name: Notification.Name(rawValue: ColorPaletteChangedNotification), object: currentColorPalette)
+        }
+    }
+    
+    public var currentColorScheme: ColorScheme = .day {
+        didSet {
+            currentColorPalette = isNight ? nightColorPalette! : dayColorPalette!
```

Comment:
Tieto `!` su safe? ak dobre vidim tak tieto 2 premenne sa nastavuju v `setDefaultColorPalettes`. Ale ak zmenit color schemu pred tym ako zavolam tu metodu tak to cele crashne nie?